engine/hooks/tasks.go

```diff
@@ -0,0 +1,98 @@
+package hooks
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ovh/cds/engine/api/workflow"
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/tat/api/cache"
+
+	"github.com/ovh/cds/sdk/log"
+)
+
+//This are all the types
+const (
+	TypeWebHook = "Webhook"
+)
+
+var (
+	longRunningRootKey          = cache.Key("hooks", "tasks", "long_running")
+	longRunningExecutionRootKey = cache.Key("hooks", "tasks", "long_running", "executions")
+	scheduledRootKey            = cache.Key("hooks", "tasks", "scheduled")
+	scheduledExecutionRootKey   = cache.Key("hooks", "tasks", "scheduled", "executions")
+
+	schedulerQueueLongRuningTasksKey = cache.Key("hooks", "scheduler", "queue", "long_running")
+	schedulerQueueScheduledTasksKey  = cache.Key("hooks", "scheduler", "queue", "scheduled")
+)
+
+// runTasks should run as a long-running goroutine
+func (s *Service) runTasks(ctx context.Context) error {
+	if err := s.synchronizeTasks(); err != nil {
+		log.Error("Hook> Unable to synchronize tasks: %v", err)
+	}
+
+	if err := s.startLongRunningTasks(ctx); err != nil {
+		log.Error("Hook> Exit long running tasks: %v", err)
```

Comment:
wrap

Reply:
no need to wrap, because this code is not called by the router, but in a single goroutine.